patches/api/0385-Add-WardenAngerEvent.patch

```diff
@@ -0,0 +1,99 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Minecaft <dustin.codes02@gmail.com>
+Date: Sun, 12 Jun 2022 00:56:45 +0200
+Subject: [PATCH] Add WardenAngerEvent
+
+Adding a event which gets called when a warden is angered by
+another entity.
+
+diff --git a/src/main/java/io/papermc/paper/event/entity/WardenAngerEvent.java b/src/main/java/io/papermc/paper/event/entity/WardenAngerEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..22578a68f59d9d87e2e42b7315149cdf2262cdd3
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/entity/WardenAngerEvent.java
+@@ -0,0 +1,85 @@
++package io.papermc.paper.event.entity;
++
++import org.bukkit.entity.Entity;
++import org.bukkit.entity.Warden;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Called when a Warden is angered by another entity.
++ * <p>
++ * If the event is cancelled, the warden will not be angered.
++ */
++public class WardenAngerEvent extends EntityEvent implements Cancellable {
++
++    private static final HandlerList handlers = new HandlerList();
++    private boolean cancelled;
++    private final Entity target;
++    private int newAnger;
++
++    public WardenAngerEvent(@NotNull final Warden warden, @Nullable final Entity target, final int newAnger) {
++        super(warden);
++        this.target = target;
++        this.newAnger = newAnger;
++    }
++
++    /**
++     * Gets the entity (if any) which triggered this anger update.
++     *
++     * @return triggering entity, or null
++     */
++    @Nullable
++    public Entity getTarget() {
++        return target;
++    }
++
++    /**
++     * Gets the new anger resulting from this event.
++     *
++     * @return new anger
++     * @see Warden#getAnger(Entity)
++     */
++    public int getNewAnger() {
++        return newAnger;
++    }
++
++    /**
++     * Sets the new anger resulting from this event.
++     *
++     * @param newAnger the new anger
++     * @see Warden#setAnger(Entity, int)
++     */
```

Comment:
I'd personally change this to anger level instead of just anger, tiny nitpick tho.